framework/contrib/hit/lex.cc

```diff
@@ -0,0 +1,386 @@
+
+#include <string>
+#include <vector>
+
+#include "lex.h"
+
+namespace hit
+{
+
+const std::string digits = "0123456789";
+const std::string alpha = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const std::string space = " \t";
+const std::string allspace = " \t\n\r";
+const std::string newline = "\n\r";
+const std::string alphanumeric = digits + alpha;
+const std::string identchars = alphanumeric + "_./:<>-+";
+
+_LexFunc::_LexFunc(LexFunc pp) : p(pp) {}
+_LexFunc::operator LexFunc() { return p; }
+
+bool
+charIn(char c, std::string valid)
```

Comment:
Any reason you can't use `const std::string &` throughout? It appears that the entire parser uses string copies everywhere. I know parsing isn't a critical region, but we typically use const references in most of MOOSE. Something to consider.